### PR TITLE
[PGO]Add `-fdiagnostics-show-profile-count` option to show real loop count from instr-profile

### PR DESCRIPTION
**Pull request by ElvisWang123**
The original `-fdiagnostics-show-hotness` option show the relative number of the loop count which is calculate by the `function_entry_count` and `branch_frequency`. We want to know the real loop iteration count in the remark which is collect in the instrument profile, adding a new option to expose the new feature.

- Add a new metadata `MD_prof_count` which contains the runtime loop iterations count. For example:
```
loop.header:
  ...
  br  i1  %0, label %true, label %false, !prof.count !0
...
!0 = !{!"profile_count", !i64 0}
```

- If option `-fdiagnostics-show-profile-count` is set we will append the `MD_prof_count` metadata at the branch instruction at the header of loops.
- Show the profile count like hotness with remark. For example:
```
  remark: the cost-model indicates that interleaving is not beneficial
  (ProfileCount: 20) [-Rpass-analysis=loop-vectorize]
  38 |   for(int i = 0;  i < argc % 20; i++){
     |   ^
```

### CODE REVIEW

File: clang/lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -2091,6 +2091,12 @@ bool CompilerInvocation::ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args,
   bool UsingProfile =
       UsingSampleProfile || !Opts.ProfileInstrumentUsePath.empty();
 
+  if (Args.hasArg(options::OPT_fdiagnostics_show_profile_count) &&
```
Comment (jansvoboda11) [inline clang/lib/Frontend/CompilerInvocation.cpp:2094]: If this option exists in `-cc1` only to emit a warning without being used for anything else, I suggest moving the diagnostic to the driver and removing the `-cc1` option.

Comment (ElvisWang123) [reply clang/lib/Frontend/CompilerInvocation.cpp:2094]: This option `options::OPT_fdiagnostics_show_profile_count` also appears in the `clang/lib/Driver/Toochains/clang.cpp` to enable the profile metadata appending to IR.

Comment (ElvisWang123) [reply clang/lib/Frontend/CompilerInvocation.cpp]: Thanks for your comment! Remove `-cc1` option and migrate warning  generation to driver.

**Comment by ElvisWang123** [comment]
Gentle ping

**Comment by ElvisWang123** [comment]
Ping

**Pull request closed without merging**